pkg/internal/gcp-get-valid-hostname.ps1

```diff
@@ -0,0 +1,25 @@
+# This script returns a mutated hostname if the length is longer than 63 characters. The hostname will be the
+# lesser of 63 characters after the first dot in the FQDN. Uses the GoogleCloud Tools for PowerShell (https://cloud.google.com/tools/powershell/docs/quickstart)
+# to get the original hostname from the GCP instance metadata service.
+
+# MAX_LENGTH is the maximum number of character allowed for the instance's hostname in GCP
+$MAX_LENGTH = 63
+
+# get hostname from the instance metadata service
+$hostname=Get-GceMetadata -Path "instance/hostname"
```

Comment:
As discussed it is best to go the internal endpoint due to the BYOH case you have pointed out. Please link to the doc I pointed you at.